mteb/evaluation/evaluators/RetrievalEvaluator.py

```diff
@@ -518,44 +516,63 @@ def evaluate(
         qrels: dict[str, dict[str, int]],
         results: dict[str, dict[str, float]],
         k_values: List[int],
-        ignore_identical_ids: bool = True,
+        ignore_identical_ids: bool = False,
     ) -> Tuple[Dict[str, float], dict[str, float], dict[str, float], dict[str, float]]:
+        logger.info(
+            "For evaluation, we do not ignore identical query and document ids (default), please explicitly set ``ignore_identical_ids=True`` to ignore this."
+        )
         if ignore_identical_ids:
-            logger.info(
-                "For evaluation, we ignore identical query and document ids (default), please explicitly set ``ignore_identical_ids=False`` to ignore this."
-            )
-            popped = []
+            results_wo_identical_ids = copy.deepcopy(results)
             for qid, rels in results.items():
                 for pid in list(rels):
                     if qid == pid:
-                        results[qid].pop(pid)
-                        popped.append(pid)
+                        results_wo_identical_ids[qid].pop(pid)
 
-        all_ndcgs, all_aps, all_recalls, all_precisions = {}, {}, {}, {}
+        all_ndcgs_no_self, all_ndcgs, all_aps, all_recalls, all_precisions = {}, {}, {}, {}, {}
 
         for k in k_values:
+            all_ndcgs_no_self[f"NDCG(no self)@{k}"] = []
             all_ndcgs[f"NDCG@{k}"] = []
             all_aps[f"MAP@{k}"] = []
             all_recalls[f"Recall@{k}"] = []
             all_precisions[f"P@{k}"] = []
 
         map_string = "map_cut." + ",".join([str(k) for k in k_values])
         ndcg_string = "ndcg_cut." + ",".join([str(k) for k in k_values])
+        ndcg_string_no_self = "ndcg_cut." + ",".join([str(k) for k in k_values])
         recall_string = "recall." + ",".join([str(k) for k in k_values])
         precision_string = "P." + ",".join([str(k) for k in k_values])
         evaluator = pytrec_eval.RelevanceEvaluator(
             qrels, {map_string, ndcg_string, recall_string, precision_string}
         )
         scores = evaluator.evaluate(results)
```

Comment:
Computing scores for a dataset where query ids and document ids are different (new default: `ignore_identical_ids=False`)